internal/zinc-compile-core/src/test/scala/sbt/internal/inc/javac/JavaCompilerSpec.scala

```diff
@@ -95,7 +95,7 @@ class JavaCompilerSpec extends UnitSpec {
         val (result, _) = compile(local, Seq(input), Seq("-d", out.getAbsolutePath))
         result shouldBe true
         val clazzz = new URLClassLoader(Array(out.toURI.toURL)).loadClass("hasstaticfinal")
-        ClassToAPI(Seq(clazzz))
+        ClassToAPI(Seq(clazzz)).sortBy(_.definitionType.hashCode)
```

Comment:
> Removing laziness from the fields of ClassLike has the side effect to randomize the order of the result 

Does the result become non-deterministic with this change? If so would that affect change detection in computing (transitive) incremental compilation? iirc, Zinc first compiles whatever that it first detected as source change, and then it subsequently uses API difference to figure out if it needs to do more work.